webapp/channels/src/components/advanced_text_editor/advanced_text_editor.test.tsx

```diff
@@ -0,0 +1,326 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {screen} from '@testing-library/react';
+import React from 'react';
+
+import type {Channel} from '@mattermost/types/channels';
+
+import Permissions from 'mattermost-redux/constants/permissions';
+
+import type {FileUpload} from 'components/file_upload/file_upload';
+import type Textbox from 'components/textbox/textbox';
+
+import WebSocketClient from 'client/web_websocket_client';
+import mergeObjects from 'packages/mattermost-redux/test/merge_objects';
+import {renderWithRealStore, userEvent} from 'tests/react_testing_utils';
+import {WebSocketContext} from 'utils/use_websocket';
+
+import type {PostDraft} from 'types/store/draft';
+
+import AdavancedTextEditor from './advanced_text_editor';
+
+global.ResizeObserver = require('resize-observer-polyfill');
+
+const currentUserId = 'current_user_id';
+const channelId = 'current_channel_id';
+
+const initialState = {
+    entities: {
+        general: {
+            config: {
+                EnableConfirmNotificationsToChannel: 'false',
+                EnableCustomGroups: 'false',
+                PostPriority: 'false',
+                ExperimentalTimezone: 'false',
+                EnableCustomEmoji: 'false',
+                AllowSyncedDrafts: 'false',
+            },
+            license: {
+                IsLicensed: 'false',
+                LDAPGroups: 'false',
+            },
+        },
+        channels: {
+            channels: {
+                current_channel_id: {
```

Comment:
You might be able to leave out some of these fields if you generate real versions of some of these objects using things like `TestHelper.getChannelMock({id: 'current_channel_id', team_id: 'current_team_id'})` in this case